src/Hangfire.PostgreSql/Factories/NpgsqlInstanceConnectionFactoryBase.cs

```diff
@@ -23,13 +24,14 @@ protected NpgsqlInstanceConnectionFactoryBase(PostgreSqlStorageOptions options)
 
   protected NpgsqlConnectionStringBuilder SetupConnectionStringBuilder(string connectionString)
   {
-    if (_connectionStringBuilder != null)
+    if (!HasConnectionStringChanged(connectionString))
```

Comment:
Negation here, negation in the method; don't exactly like this.

I suggest not even adding the method and doing the comparison directly in the condition. Null-check of builder should stay as well as a first check for short circuit.

Reply:
Have addressed your review comment.